vue-begin/src/components/Employees.vue

```diff
@@ -0,0 +1,187 @@
+      <v-card-text>
+        <v-container grid-list-md>
+          <v-layout wrap>
+            <v-flex xs12 sm6>
+              <v-text-field 
+                label="Employee name"
+                v-model="dialogForm.dialogName" 
+                required/>
+            </v-flex>
+            <v-flex xs12 sm6>
+              <v-text-field
+                label="Salary"
+                type="number"
+                min="0"
+                v-model="dialogForm.dialogSalary" 
+                required>
+              </v-text-field>
+            </v-flex>
+            <v-flex xs12 sm6>
+              <v-text-field 
+                label="Age"
+                type="number"
+                min="0"
+                max="120"
+                v-model="dialogForm.dialogAge" 
+                required>
+              </v-text-field>
+            </v-flex>
+          </v-layout>
+        </v-container>
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn color="blue darken-1" flat
+          @click="showDialog(false)">Close</v-btn>
+        <v-btn color="blue darken-1" flat
+         v-if="dialogForm.dialogId" @click="add()">Add</v-btn>
+         <v-btn color="blue darken-1" flat
+         v-else @click="update(dialogForm.dialogId)">Update</v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+  </div>
+</template>
+
+<script>
+  import * as APIService from '../APIService';
+
+const createDefaultItem = () => ({
+  dialogName: '',
+  dialogSalary: '',
+  dialogShow: false,
+  dialogId: true,
+  dialogAge: null,
+})
+
+  export default {
+    name: 'Employees',
+    props: {
+      msg: String
+    },
+    data() {
+        return {
+            employees: [],
+            employeesPerPage: 20,
+            currentPage: 1,
+            dialogShow: false,
+            dialogForm:{
+              dialogId: null,
+              dialogName: null,
+              dialogSalary: null,
+              dialogAge: null,
+            }
+        };
+    },
+
+    computed: {
+      paginatedEmployees() {
+        return this.employees.slice(
+          this.currentPage * this.employeesPerPage, (this.currentPage + 1) * this.employeesPerPage - 1);
+      }
+    },
+
+    mounted() {
+      this.load();
+    },
+
+    methods: {
+      async load() {
+        this.employees = await APIService.getEmployees();
+      },
+
+      showDialog(show, item = createDefaultItem()  ) {
+        this.dialogShow = show;
+        this.dialogForm = {...item};
+        // this.dialogForm.dialogId = (item || {}).id;
+        // this.dialogForm.dialogName = (item || {}).name;
+        // this.dialogForm.dialogSalary = (item || {}).salary;
+        // this.dialogForm.dialogAge = (item || {}).age;
+      },
+
+      add() {
+        let payload = {
+          name: this.dialogName,
+          salary: this.dialogSalary,
+          age: this.dialogAge
+        }
+        console.log(name);
+        APIService.createEmployee(data,payload);
```

Comment:
`createEmployee` is supposed to get only payload parameter.

Reply:
ok